Userland/Libraries/LibWebView/OutOfProcessWebView.cpp

```diff
@@ -295,6 +328,54 @@ void OutOfProcessWebView::enqueue_native_event(Web::MouseEvent::Type type, GUI::
     enqueue_input_event(Web::MouseEvent { type, position, screen_position, static_cast<Web::UIEvents::MouseButton>(to_underlying(event.button())), static_cast<Web::UIEvents::MouseButton>(event.buttons()), static_cast<KeyModifier>(event.modifiers()), wheel_delta_x, wheel_delta_y, nullptr });
 }
 
+struct DragData : Web::ChromeInputData {
+    explicit DragData(GUI::DropEvent const& event)
+        : event(make<GUI::DropEvent>(event))
+    {
+    }
+
+    NonnullOwnPtr<GUI::DropEvent> event;
+};
+
+void OutOfProcessWebView::enqueue_native_event(Web::DragEvent::Type type, GUI::DropEvent const& event)
+{
+    auto position = to_content_position(event.position()).to_type<Web::DevicePixels>();
+    auto screen_position = (event.position() + (window()->position() + relative_position())).to_type<Web::DevicePixels>();
+
+    Vector<Web::HTML::SelectedFile> files;
+    OwnPtr<DragData> chrome_data;
+
+    auto button = static_cast<Web::UIEvents::MouseButton>(to_underlying(event.button()));
+    auto buttons = static_cast<Web::UIEvents::MouseButton>(event.buttons());
+    auto modifiers = static_cast<Web::UIEvents::KeyModifier>(event.modifiers());
```

Comment:
`Web::UIEvents::KeyModifier` is the same type at the moment, abut `Web::UIEvents::MouseButton` and `GUI::MouseButton` are distinct and just happen to line up. I suppose that's alright, but it's a bit wobbly :)

Reply:
Yeah, I had just copied these from the mouse event code above, but I agree, I'd rather these be explicit. I added a commit to add conversions & use them in mouse events.